### PR TITLE
[FIX] whitespace: validate js upon whitespace modification

### PR DESCRIPTION
**Pull request by tobiasso85**
When modifying whitespace to improve the formatting.
JS code is validated for each modification, to ensure
that the code doesn't break.

fixes: #121

### CODE REVIEW

File: src/util/whitespace/AstStringOptimizeStrategy.ts [hunk 1]
```diff
@@ -341,6 +341,20 @@ export class AstStringOptimizeStrategy implements StringOptimizeStrategy {
 		return result;
 	}
 
+	/**
+	 *
```
Comment (svbender) [inline src/util/whitespace/AstStringOptimizeStrategy.ts:345]: Description?

File: src/util/whitespace/AstStringOptimizeStrategy.ts [hunk 2]
```diff
@@ -356,42 +370,79 @@ export class AstStringOptimizeStrategy implements StringOptimizeStrategy {
...
-				const iIndexToDelete =
-					iIndex - (whitespaceToRemove.length - 1 - i);
-				const sCharInMod = aOptimizedContent[iIndexToDelete];
-				if (sCharInMod === sWhitespaceChar) {
-					aOptimizedContent[iIndexToDelete] = "";
-					lastDeletedIndex = iIndexToDelete;
+		const aOptimizedContentBackup = aOptimizedContent.slice();
+
+		/**
+		 * collects actions to perform
```
Comment (svbender) [inline src/util/whitespace/AstStringOptimizeStrategy.ts:376]: Maybe "Collects actions to perform a simulation of a modification and test it..." ???

This is hard to understand.. what is an action for? What is tested and why? An example could be useful.

File: src/util/whitespace/AstStringOptimizeStrategy.ts (hunk 1 above)
Comment (tobiasso85) [reply src/util/whitespace/AstStringOptimizeStrategy.ts:345]: done

File: src/util/whitespace/AstStringOptimizeStrategy.ts (hunk 2 above)
Comment (tobiasso85) [reply src/util/whitespace/AstStringOptimizeStrategy.ts]: done

**Pull request merged**